File: sdk/src/test/java/com/cxense/cxensesdk/model/ExternalUserIdTest.java
package com.cxense.cxensesdk.model;

import com.cxense.cxensesdk.BaseTest;

import org.junit.Test;

public class ExternalUserIdTest extends BaseTest {

    @Test
    public void createWithInternalId() {
        new ExternalUserId("cx", "test");
    }

    @Test(expected = IllegalArgumentException.class)
    public void createWithInternalIdError() {
        new ExternalUserId("cx", "test/");
    }
    @Test
    public void createWithExternalId() {
        new ExternalUserId("cxd", "test/");
    }

    @Test(expected = IllegalArgumentException.class)
    public void createWithExternalIdError() {
        new ExternalUserId("cxd", "test'");
    }
}

File: sdk/src/test/java/io/piano/android/cxense/model/ExternalUserIdTest.kt
package io.piano.android.cxense.model

import io.piano.android.cxense.assertFailsWithMessage
import kotlin.test.Test

class ExternalUserIdTest {
    @Test
    fun createWithInternalId() {
        ExternalUserId("cx", "test")
    }

    @Test
    fun createWithInternalIdError() {
        assertFailsWithMessage<IllegalArgumentException>(
            "characters for internal ids",
            "Expected fail for internal user id"
        ) {
            ExternalUserId("cx", "test/")
        }
    }

    @Test
    fun createWithExternalId() {
        ExternalUserId("cxd", "test/")
    }

    @Test
    fun createWithExternalIdError() {
        assertFailsWithMessage<IllegalArgumentException>(
            "characters for external ids",
            "Expected fail for external user id"
        ) {
            ExternalUserId("cxd", "test'")
        }
    }
}
